tests/0101-fetch-from-follower.cpp

```diff
@@ -0,0 +1,305 @@
+ *  - consume the messages, and check they are as expected.
+ *  - use rxbytes from the statistics event to confirm that 
+ *    the messages were retrieved from the replica broker (not the
+ *    leader).
+ */
+
+
+static void test_assert(bool cond, std::string msg) {
+  if (!cond)
+    Test::Say(msg);
+  assert(cond);
+}
+
+
+class TestEvent2Cb : public RdKafka::EventCb {
+ public:
+  static bool should_capture_stats;
+  static bool has_captured_stats;
+  static std::map<int32_t, int64_t> rxbytes;
+
+  void event_cb (RdKafka::Event &event) {
+
+    switch (event.type())
+    {
+      case RdKafka::Event::EVENT_LOG:
+        Test::Say(event.str() + "\n");
+        break;
+      case RdKafka::Event::EVENT_STATS:
+        if (should_capture_stats) {
+
+          rapidjson::Document d;
+          if (d.Parse(event.str().c_str()).HasParseError())
+            Test::Fail(tostr() << "Failed to parse stats JSON: " <<
+                       rapidjson::GetParseError_En(d.GetParseError()) <<
+                       " at " << d.GetErrorOffset());
+
+          /* iterate over brokers. */
+          rapidjson::Pointer jpath((const char *)"/brokers");
+          rapidjson::Value *pp = rapidjson::GetValueByPointer(d, jpath);
+          if (pp == NULL)
+            return;
+
+          for (rapidjson::Value::ConstMemberIterator itr = pp->MemberBegin(); itr != pp->MemberEnd(); ++itr) {
+            std::string broker_name = itr->name.GetString();
+            size_t broker_id_idx = broker_name.find('/');
+            if (broker_id_idx == (size_t)-1)
+              continue;
+            std::string broker_id = broker_name.substr(broker_id_idx + 1, broker_name.size() - broker_id_idx - 1);
+
+            int64_t broker_rxbytes = itr->value.FindMember("rxbytes")->value.GetInt64();
+            rxbytes[std::stoi(broker_id)] = broker_rxbytes;
+          }
+
+          has_captured_stats = true;
+          break;
+        }
+      default:
+        break;
+    }
+  }
+};
+
+bool TestEvent2Cb::should_capture_stats;
+bool TestEvent2Cb::has_captured_stats;
+std::map<int32_t, int64_t> TestEvent2Cb::rxbytes;
+static TestEvent2Cb ex_event_cb;
+
+
+static void get_brokers_info(std::string &topic_str, int *leader, std::vector<int> &brokers) {
+  std::string errstr;
+  RdKafka::ErrorCode err;
+  class RdKafka::Metadata *metadata;
+
+  /* Determine the ids of the brokers that the partition has replicas
+   * on and which one of those is the leader.
+   */
+  RdKafka::Conf *pConf;
+  Test::conf_init(&pConf, NULL, 10);
+  RdKafka::Producer *p = RdKafka::Producer::create(pConf, errstr);
+  delete pConf;
+  test_assert(p, tostr() << "Failed to create producer: " << errstr);
+
+  RdKafka::Conf *tconf = RdKafka::Conf::create(RdKafka::Conf::CONF_TOPIC);
+  RdKafka::Topic *topic = RdKafka::Topic::create(p, topic_str, tconf, errstr);
+  delete tconf;
+  test_assert(topic, tostr() << "Failed to create topic: " << errstr);
+
+  err = p->metadata(0, topic, &metadata, 5000);
```

Comment:
use `tmout_multip(5000)` which gives code like this more time on CI, under valgrind, etc.